contracts/Database.sol

```diff
@@ -43,9 +51,41 @@ contract Database {
         return _id;
     }
 
+    function _checkGateName(string memory _name) private view {
```

Comment:
Also change this to a mapping(string => uint256) and just check if it exists

Reply:
as in, keep a new gate name mapping for each merchant? 

Is the extra storage less expensive than the extra computation in this case?